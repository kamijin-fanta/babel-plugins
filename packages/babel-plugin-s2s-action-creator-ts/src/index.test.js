import plugin from './index';
import pluginTester from 'babel-plugin-tester';
import path from 'path';

const filename = path.resolve(process.cwd(), 'app', 'actionTypes.ts');

pluginTester({
  plugin,
  pluginOptions: {
  },
  babelOptions: {
    filename,
  },
  snapshot: true,
  tests: [
    {
      title: 'usePrefix: false',
      code: `
        import External from 'external';
        export type Action = Hoge | Fuga;
        
        export interface Hoge {
          type: Actions.Hoge;
          key: number;
          data: External;
        }
        export interface Fuga {
          type: Actions.Fuga;
        }
        
        /***** Do not edit below this line *****/
        export const enum Actions {
          Hoge = 'containers/Hoge',
          Fuga = 'containers/Fuga',
        }
      `,
    },
  ],
});
