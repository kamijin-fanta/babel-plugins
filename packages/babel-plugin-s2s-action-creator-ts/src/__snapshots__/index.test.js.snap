// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`usePrefix: false 1`] = `
"
import External from 'external';
export type Action = Hoge | Fuga;

export interface Hoge {
  type: Actions.Hoge;
  key: number;
  data: External;
}
export interface Fuga {
  type: Actions.Fuga;
}

/***** Do not edit below this line *****/
export const enum Actions {
  Hoge = 'containers/Hoge',
  Fuga = 'containers/Fuga',
}

      ↓ ↓ ↓ ↓ ↓ ↓

/***** Do not edit this file *****/
import External from 'external';
import { Actions, Hoge, Fuga } from \\"./actionTypes\\";
export function hoge(key: number, data: External) {
  return {
    type: Actions.Hoge,
    key,
    data
  };
}
export function fuga() {
  return {
    type: Actions.Fuga
  };
}
"
`;
